fedimint-server/src/consensus/mod.rs

```diff
@@ -615,6 +617,39 @@ impl FedimintConsensus {
         }
     }
 
+    pub async fn download_config_with_token(
+        &self,
+        info: WsClientConnectInfo,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
+    ) -> ApiResult<ConfigResponse> {
+        let token = self.cfg.local.download_token.clone();
+
+        if info.download_token != token {
+            return Err(ApiError::bad_request(
+                "Download token not found".to_string(),
+            ));
+        }
+
+        let times_used = dbtx
+            .get_value(&ClientConfigDownloadKey(token.clone()))
+            .await
+            .unwrap_or_default();
+
+        dbtx.insert_entry(&ClientConfigDownloadKey(token), &(times_used + 1))
```

Comment:
Do we want to increment `times_used` here before checking if its over the limit? I haven't read what the caller is doing yet, but if we're at 101 times used, then this API is called, the request will be rejected but the value will be increment unnecessarily to 102 right?